packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -1567,3 +1577,83 @@ class TextStyle with Diagnosticable {
     styles.add(EnumProperty<TextOverflow>('${prefix}overflow', overflow, defaultValue: null));
   }
 }
+
+/// Interpolate between two lists of [FontVariation] objects.
+///
+/// Variations are paired by axis, and interpolated using [FontVariation.lerp].
+///
+/// This algorithm is O(N), but the constant factor varies based on the input.
+///
+/// It is fastest when the lists contain the same axes ([FontVariation.axis]) in
+/// the same order. The result is again in the same order, and no attempt is
+/// made to detect or remove duplicates in this process.
+///
+/// When the order of items in the two input lists vary, the constant factor
+/// increases substantially, as it involves creating two maps and a set,
+/// inserting every variation in both lists into the maps and the set, and then
+/// iterating over them to recreate the list.
+///
+/// In this case, the resulting order is arbitrary. Duplicates are dropped; in
```

Comment:
If we want to assert on duplicates, that's probably something for the engine side. It would be weird to assert here, since there's nothing about the API today that disallows asserts.

I don't have a strong opinion on whether duplicates should be allowed or not.

Reply:
yeah that's probably the reason we went with this design